crypto/x509/x509_cmp.c

```diff
@@ -160,7 +162,7 @@ int X509_cmp(const X509 *a, const X509 *b)
         return memcmp(a->cert_info.enc.enc, b->cert_info.enc.enc,
                       a->cert_info.enc.len);
     }
-    return rv;
+    return rv < 0 ? -1 : rv > 0;
```

Comment:
just return 0 here.

Reply:
Well, in this case it would be sufficient to return 0,
but I just realized that I forgot to transform
```
return memcmp(a->cert_info.enc.enc, b->cert_info.enc.enc, a->cert_info.enc.len);
```
to 
```
rv = memcmp(a->cert_info.enc.enc, b->cert_info.enc.enc, a->cert_info.enc.len);
```
I've just done this, and now the `rv < 0 ? -1 : rv > 0` is actually needed.